soda/core/soda/execution/data_source.py

```diff
@@ -488,26 +488,35 @@ def profiling_sql_aggregates_text(self, table_name: str, column_name: str) -> st
         )
 
     def histogram_sql_and_boundaries(
-        self, table_name: str, column_name: str, min: int | float, max: int | float
+        self,
+        table_name: str,
+        column_name: str,
+        minimum: int | float,
+        maximum: int | float,
+        n_distinct: int,
+        column_type: str,
     ) -> tuple[str | None, list[int | float]]:
         # TODO: make configurable or derive dynamically based on data quantiles etc.
-        number_of_bins: int = 20
+        max_n_bins = 20
+        number_of_bins: int = max(1, min(n_distinct, max_n_bins))
+        number_of_intervals: int = number_of_bins - 1
 
-        if not min < max:
+        if not minimum < maximum:
             self.logs.warning(
-                f"Min of {column_name} on table: {table_name} must be smaller than max value. Min is {min}, and max is {max}"
+                f"Min of {column_name} on table: {table_name} must be smaller than max value. Min is {minimum}, and max is {maximum}"
             )
             return None, []
 
-        min_value = floor(min * 1000) / 1000
-        max_value = ceil(max * 1000) / 1000
-        bin_width = (max_value - min_value) / number_of_bins
+        min_value = floor(minimum * 1000) / 1000
```

Comment:
@bastienboutonnet why do we need floor and ceil in these lines?

Reply:
I literally have no idea, I stole the code from soda-sql to maintain feature parity and consistency with derivation and didn't take the time to properly evaluate the soundness.

Do you have an issue with it?

Reply:
I would say if it's unsound, maybe let's open up a separate issue for this and tackle it then, because if we change the way things are derived it may cause unexpected differences and I would prefer that we have an explanation and potentially an annoucement or release note for it.

Reply:
nope just wondering whether we should keep it or not :D I think deleting shouldn't cause any issue since we already round the bins

Reply:
yeah... seems like so. Please open a ticket so we can prio it and check whether it leads to differences or not and if not, we can remove it because unnecessary code isn't nice to have.